### PR TITLE
Defaults and deps

### PR DESCRIPTION
**Pull request by nickbalestra**
- added defaults
- added tests
- moved dependencies to middleware logic (self registering serveStatic, due to path issues when requiring the falcor-postman as an npm installed module)

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -69,4 +69,8 @@ App.propTypes = {
   falcorPath: React.PropTypes.string,
 }
 
+App.defaultProps = {
+  falcorPath: '/mode.json'
```
Comment (mattiaerre) [inline app/app.js]: there is a typo man ;)

File: middleware.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import { renderToString as render } from 'react-dom/server'
 import React from 'react'
 import App from './app/app'
+import express from 'express'
+import path from 'path'
 
-module.exports = (options) =>
-  (req, res, next) => {
-    if (req.url === options.middlewarePath) {
+module.exports = ({middlewarePath='/falcor-postman', app}) => {
```
Comment (mattiaerre) [inline middleware.js:7]: do we want to inject this value as well?

File: test/middleware.spec.js [hunk 3]
```diff
@@ -21,27 +31,54 @@ describe('middleware', () => {
...
 
-      it('then next called', () => {
+      it('should call next', () => {
         sinon.assert.called(next)
       })
     })
 
-    describe('when invoked and options.middlewarePath matches req.url', () => {
-      const req = { url: options.middlewarePath }
+    describe('when invoked at default /falcor-postman (and options.middlewarePath is not defined) matches req.url', () => {
```
Comment (mattiaerre) [inline test/middleware.spec.js:39]: splendid 👍

**Review by mattiaerre — COMMENTED** [review]
over to you @nickbalestra

**Comment by nickbalestra** [comment]
@mattiaerre I configured travis, now up and running properly, I'll leave it to you for review/merge

File: middleware.js (hunk 2 above)
Comment (mattiaerre) [reply middleware.js:7]: 😅

**Review by mattiaerre — COMMENTED** [review]
LGTM 👍

**Pull request merged**